rust/lance-core/src/utils/tracing.rs

```diff
@@ -75,4 +75,11 @@ pub const IO_TYPE_OPEN_MEM_WAL: &str = "open_mem_wal_index";
 pub const IO_TYPE_LOAD_VECTOR_PART: &str = "load_vector_part";
 pub const IO_TYPE_LOAD_SCALAR_PART: &str = "load_scalar_part";
 pub const TRACE_EXECUTION: &str = "lance::execution";
+pub const TRACE_DATASET_EVENTS: &str = "lance::dataset_events";
+pub const DATASET_EVENT_CREATE_OR_WRITE: &str = "dataset_create_or_write";
```

Comment:
We may track `fragment` and `data file` related events in the future. So I added `dataset_` prefix.

Reply:
but you can always know the event is for a dataset from the `target = lance::dataset_events`?